File: php/features.php
<?php
declare(strict_types=1);

// Openfoodfacts SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class OpenfoodfactsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new OpenfoodfactsBaseFeature();
            case "test":
                return new OpenfoodfactsTestFeature();
            default:
                return new OpenfoodfactsBaseFeature();
        }
    }
}

File: py/openfoodfacts_sdk/features.py
# Openfoodfacts SDK feature factory

from openfoodfacts_sdk.feature.base_feature import OpenfoodfactsBaseFeature
from openfoodfacts_sdk.feature.test_feature import OpenfoodfactsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: OpenfoodfactsBaseFeature(),
        "test": lambda: OpenfoodfactsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
